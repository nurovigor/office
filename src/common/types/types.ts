export type OpenSpaceNumber = 1 | 2 | 3 | 4;

export type LinkType = {
	path: string;
	title: string;
};

export type RotateType = 0 | 90 | 180 | 270;

export type DirectionType = 'top' | 'right' | 'bottom' | 'left';

export type Nullable<T> = null | T;

export type FilterName = {
	name: string;
	type: string;
	bind: string;
};

export type UpdateDataType = {
	developerId: Nullable<string>;
	tableId: Nullable<string>;
	pc: Nullable<string>;
	monitor: Nullable<string>;
	keyboard: Nullable<string>;
	mouse: Nullable<string>;
	microphone: Nullable<string>;
	headphones: Nullable<string>;
	camera: Nullable<string>;
};

export type GetTechnics = {
	name: string;
	type: string;
	bind: string;
	currentPage: number;
	pageSize: number;
	sort: string[];
};

export interface TechnicTypeI {
	bind: string;
	name: Nullable<string>;
	serial: Nullable<string>;
	type: Nullable<string>;
	__v: number;
	_id: string;
}

export type FilterType = {
	name: string;
	selectedOption: string;
	options: string[];
};

export type CreateDataResponse = {
	name: string;
	type: string;
	serial: string;
	bind: string;
};
