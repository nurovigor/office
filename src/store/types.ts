import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { store } from './store';

//types
export type AppStateType = ReturnType<typeof store.getState>;
//AppState type
export type AppDispatch = ThunkDispatch<AppStateType, unknown, AppActionType>;
//thunks type
export type AppThunk<ReturnType = void> = ThunkAction<
	ReturnType,
	AppStateType,
	unknown,
	AppActionType
>;
//actions type
export type AppActionType = any;
