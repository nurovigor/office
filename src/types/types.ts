export type OpenSpaceNumber = 1 | 2 | 3 | 4;

export type LinkType = {
	path: string;
	title: string;
};

export type RotateType = 0 | 90 | 180 | 270;

export type DirectionType = 'top' | 'right' | 'bottom' | 'left';

export type Nullable<T> = null | T;
