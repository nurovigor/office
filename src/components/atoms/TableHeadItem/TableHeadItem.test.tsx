import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { TableHeadItem } from './';
import { headData } from 'src/common/types/types';

describe('TableHeadItem component', () => {
	let item: headData[];
	let sortValues: string[];
	let onClick: any;
	beforeEach(() => {
		item = [
			{
				id: '1',
				name: 'name',
				accessor: 'name'
			},
			{
				id: '2',
				name: 'type',
				accessor: 'type'
			},
			{
				id: '3',
				name: 'bind',
				accessor: 'bind'
			}
		];
		sortValues = ['name'];

		onClick = jest.fn();
	});
	it('renders TableHeadItem component', () => {
		const { container } = render(
			<table>
				<TableHeadItem item={item} sortValues={sortValues} handleSorting={onClick} />
			</table>
		);
		expect(container).toMatchSnapshot();
	});

	/*	it('renders with classname UP', () => {
		render(
			<table>
				<TableHeadItem item={item} sortValues={sortValues} handleSorting={onClick} />
			</table>
		);
		expect(screen.getAllByTestId('arrow-filter')[0]).toHaveClass('Sort');
	});*/

	it('onClick works', () => {
		render(
			<table>
				<TableHeadItem item={item} sortValues={sortValues} handleSorting={onClick} />
			</table>
		);
		fireEvent(
			screen.getAllByRole('button')[0],
			new MouseEvent('click', {
				bubbles: true,
				cancelable: true
			})
		);
		expect(onClick).toBeCalledWith('name', 'asc');
	});
});
