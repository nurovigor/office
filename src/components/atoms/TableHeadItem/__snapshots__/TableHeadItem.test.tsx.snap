// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadItem component renders TableHeadItem component 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th
          title="name"
        >
          <button
            class=""
            type="button"
          >
            <span />
            name
          </button>
          <div
            data-testid="arrow-filter"
          />
        </th>
        <th
          title="type"
        >
          <button
            class=""
            type="button"
          >
            <span />
            type
          </button>
          <div
            data-testid="arrow-filter"
          />
        </th>
        <th
          title="bind"
        >
          <button
            class=""
            type="button"
          >
            <span />
            bind
          </button>
          <div
            data-testid="arrow-filter"
          />
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
